core/modules/query/FuncExpr.h

```diff
@@ -63,15 +64,22 @@ class FuncExpr {
     static FuncExpr::Ptr newArg1(std::string const& newName,
                                  ValueExprPtr ve);
 
+    void setName(const std::string& val);
+
     void findColumnRefs(ColumnRef::Vector& outputRefs);
     std::shared_ptr<FuncExpr> clone() const;
 
     // Fields
-    std::string name;
     ValueExprPtrVector params;
```

Comment:
Can we make all attributes private?

Reply:
yes; can I do it on its own ticket? is it worth mucking about with our IR so much? (@fritzm )